### ISSUE TITLE
Suggestion: implement automatic login via NTLM challenge when AD authentication is enabled

### ISSUE THREAD

**Issue by dsteinweg** [continuousit/seq-releases#383]
We just installed a trial of the Business version of Seq, and enabled AD integration.  It worked flawlessly, but there were two things I noticed:

1. Upon visiting Seq for the first time, you're presented with a log in screen.  This is a somewhat less ideal experience when compared to other AD-authenticated applications which handle the challenge and log in for you, automatically.  Fortunately, after logging in, your session persists via a cookie.
2. Upon submitting the log in form, your provided account name and password are submitted in the POST request back to Seq--in plain text.  This practically requires that Seq is secured behind SSL, because otherwise your account credentials would be visible in the clear over HTTP.

If the Seq web server supported Windows authentication via the NTLM challenge-response process, both items above would be alleviated.  However, I *think* that the NTLM negotiate process is only supported on IIS, and since Seq doesn't use IIS, this would be an implementation challenge.  Unless there's a way to utilize IIS' Windows Authentication Module outside of IIS, this may require hosting Seq on IIS, which would likely require significant redesign.  If it were possible though, perhaps the IIS hosting option is a separate installer from the normal self-hosted web server/Windows service.

Thanks for the consideration!  Loving the product.

**Comment by nblumhardt** [comment]
Thanks Darren, great to hear! :-)

Yes, SSL is required if you want to run Seq securely in a networked situation. Plain HTTP means no protection for log event data and other potentially sensitive information that might travel over the wire. so login details are only one facet of this.

Switching to NTLM challenge/response sounds like a great idea nonetheless, definitely something we can look into to remove a bit of friction :+1:

**Comment by dsteinweg** [comment]
Nicholas, thanks for the quick response and for considering my suggestion!

_Extremely_ coincidentally, I was having a look at Octopus Deploy today, and their app's sign in page has this exact functionality!  You're presented with a form, but there's a separate link you can click which will perform the NTLM challenge/response like I'd described above.

In discovering this, I searched for "Octopus Deploy" and "AD authentication" and discovered [some sample code](https://github.com/OctopusDeploy/Issues/issues/807) on how it could work.  I, too, had assumed that Windows auth was implemented within IIS, but Paul Stovell at Octopus proved otherwise--it's in HTTP.sys, and I imagine should be usable from within your Nancy server.

(It seems you have some familiarity with Octopus Deploy, since you've got some GitHub activity yourself in that issue.  Small world!)

**Comment by nblumhardt** [comment]
Thanks for the link Darren, that's ticket I'd intended to Google for when I have time to dig into this :-) Yes, this is applicable and would be the path we take. I'll keep you posted!

**Comment by nblumhardt** [comment]
![image](https://cloud.githubusercontent.com/assets/342712/14103735/d4a0943e-f5e4-11e5-8c44-2ed2e95cd9c3.png)

**Comment by nblumhardt** [comment]
Done.

**Comment by nblumhardt** [comment]
Seq [3.1.1-pre](https://getseq.net/Download) includes this, it'd be great to have feedback if anyone is able to check this out.

**Comment by dsteinweg** [comment]
First, thank you so much for your quick responses and turnaround on the implementation!  It blew away my expectations, although I'll be careful to not expect this speed in the future :).  We just purchased a Business license, and although we would have done it regardless, this was a wonderful exchange from a customer's point of view.  Thanks!

I downloaded 3.1.1-pre and installed it on my local Windows 7 machine, and also on a dev server running Server 2008 R2.  Got a second trial key for the Business license and applied it to both instances.  Clicking that "Integrated Windows Authentication" link resulted in that wonderful auto-negotiation, and I was logged into Seq just like that!  And the HTTP requests no longer contained my username nor password, which is perfect.

I think I'll wait for the full 3.1.1 release before installing it on our _prod _Seq server, but from what I've seen, this feature in this prelease seems to work very well.  Thanks Nicholas!
